### DISCUSSION TITLE
Cant clone repo with Eclipse

### DISCUSSION THREAD

**Discussion by mommaroodles** [community/community#44393] [New to GitHub]
I'm at my wits end - I cannot believe that for something seemingly so simple - it has to turn out over complicated - surely somewhere on the web  is clear instructions as to how to generate ssh keys add them.

I eventually after a few days managed to get the SSH keys generated and added to the SSH Agent I then added the key to the relevant repo - Now my understanding is that the private key matching the public key - as that is what Eclipse asks for - must also be added in Eclipse - then further more I would need to use the following url git@github.com/xxxxxxxxxxxxxx.git in order to clone the repo using Eclipse but nothing is working I'm getting connection timed out errors, cant login and a few others. Just nothing works!!

**Comment by rashidwassan** [comment]
It sounds like you have completed the necessary steps to set up SSH key authentication with GitHub, but you are still having issues cloning the repository in Eclipse.

Here are a few things you can try:

- Make sure that the URL you are using in Eclipse is the SSH URL for the repository, not the HTTPS URL.
- Check that your internet connection is stable and that you have no firewall or proxy settings blocking your connection to GitHub.
- Confirm that you have added the private key to Eclipse by going to Window > Preferences > SSH2 > Key Management and checking that the key is listed there.
- Try to clone the repository from the command line to see if you can identify the problem.
- Check the troubleshooting section in eclipse documentation for more information.

If none of these solutions work, please provide more information about the specific error messages.

**Comment by jge162** [comment]
It sounds like you've taken the correct steps so far by generating the SSH keys, adding them to the SSH agent and adding the public key to the relevant repository. The next step would be to add the private key in Eclipse by going to the settings or preferences and navigating to the SSH section. After that, you should be able to use the git URL [git@github.com](mailto:git@github.com)/xxxxxxxxxxxxxx.git to clone the repository using Eclipse.

If you're still experiencing connection time out errors, unable to login, and other issues, it's possible that there is a problem with the configurations of your SSH keys or with the git URL you are using. It may be helpful to double-check that the private and public keys match, that you've entered the correct git URL and that the repository is correctly configured to accept your key. Additionally, you can try running ssh -T [git@github.com](mailto:git@github.com) on the command line to see if there's an issue with your SSH setup.

If all else fails, it may be helpful to seek help from the community or the support team of the platform you are using, they may be able to help troubleshoot the issue and provide more specific solutions.
